djautotask/sync.py

```diff
@@ -340,6 +340,66 @@ def _get_query_conditions(self, query):
         return query
 
 
+class BatchQueryMixin:
+    """
+    Fetch records from the API only fetching records if the related model(s)
+    are already in the database.
+    """
+
+    def __init__(self, *args, **kwargs):
+        settings = DjautotaskSettings().get_settings()
+        self.batch_size = settings.get('batch_query_size')
+        super().__init__(*args, **kwargs)
+
+    def build_batch_queries(self, sync_job_qset):
+        raise NotImplementedError
+
+    def get(self, results):
+
+        sync_job_qset = models.SyncJob.objects.filter(
+            entity_name=self.model_class.__name__
+        )
+        object_queries = self.build_batch_queries(sync_job_qset)
+
+        # Apply extra conditions if they exist, else nothing happens
+        for query in object_queries:
+            self._get_query_conditions(query)
```

Comment:
I just realized this can be moved into the loop with fetch records, just before the fetch. I'll do that soon, still ready for review otherwise.